src/platform/Tizen/AppPreference.cpp

```diff
@@ -19,145 +19,106 @@
 #include <app_preference.h>
 #include <lib/support/Base64.h>
 #include <lib/support/CHIPMem.h>
+#include <memory>
 
 namespace chip {
 namespace DeviceLayer {
 namespace PersistedStorage {
 namespace Internal {
 namespace AppPreference {
 
-static CHIP_ERROR __IsKeyExist(const char * key)
-{
-    CHIP_ERROR err = CHIP_NO_ERROR;
-    int preErr     = PREFERENCE_ERROR_NONE;
-    bool isExist   = false;
-
-    preErr = preference_is_existing(key, &isExist);
-    if (preErr != PREFERENCE_ERROR_NONE)
-    {
-        err = CHIP_ERROR_INCORRECT_STATE;
-    }
-    else if (isExist == false)
-    {
-        err = CHIP_ERROR_PERSISTED_STORAGE_VALUE_NOT_FOUND;
-    }
-
-    return err;
-}
-
 CHIP_ERROR CheckData(const char * key)
 {
-    return __IsKeyExist(key);
+    bool isExist = false;
+    int err      = preference_is_existing(key, &isExist);
+    VerifyOrReturnError(err == PREFERENCE_ERROR_NONE, CHIP_ERROR_INCORRECT_STATE);
+    return isExist ? CHIP_NO_ERROR : CHIP_ERROR_PERSISTED_STORAGE_VALUE_NOT_FOUND;
 }
 
 CHIP_ERROR GetData(const char * key, void * data, size_t dataSize, size_t * getDataSize, size_t offset)
 {
-    CHIP_ERROR err                = CHIP_NO_ERROR;
-    int preErr                    = PREFERENCE_ERROR_NONE;
-    char * encodedData            = NULL;
-    uint8_t * decodedData         = NULL;
-    size_t encodedDataSize        = 0;
-    size_t encodedDataPaddingSize = 0;
-    size_t decodedDataSize        = 0;
-    size_t expectedDecodedSize    = 0;
-    size_t copy_size              = 0;
-
-    VerifyOrExit(data != NULL, err = CHIP_ERROR_INVALID_ARGUMENT);
-
-    err = __IsKeyExist(key);
-    VerifyOrExit(err == CHIP_NO_ERROR, ChipLogProgress(DeviceLayer, "Not found data [%s]", key));
-
-    preErr = preference_get_string(key, &encodedData);
-    VerifyOrExit(preErr == PREFERENCE_ERROR_NONE, err = CHIP_ERROR_INCORRECT_STATE);
-    encodedDataSize = strlen(encodedData);
-
-    if ((encodedDataSize > 0) && (encodedData[encodedDataSize - 1] == '='))
-    {
-        encodedDataPaddingSize++;
-        if ((encodedDataSize > 1) && (encodedData[encodedDataSize - 2] == '='))
-            encodedDataPaddingSize++;
-    }
-    expectedDecodedSize = ((encodedDataSize - encodedDataPaddingSize) * 3) / 4;
-
-    decodedData = static_cast<uint8_t *>(chip::Platform::MemoryAlloc(expectedDecodedSize));
-    VerifyOrExit(decodedData != NULL, err = CHIP_ERROR_INCORRECT_STATE);
+    char * encodedData = nullptr;
 
-    decodedDataSize = Base64Decode(encodedData, static_cast<uint16_t>(encodedDataSize), decodedData);
+    VerifyOrReturnError(data != nullptr, CHIP_ERROR_INVALID_ARGUMENT);
 
```

Comment:
Should we call CheckData(key) before getting its value?

Reply:
To be honest I'm not a big fan checking every function parameter. The check for data was already there so I've left it here. Linux platform implementation does not check for "key" being not null, so I think that it's safe to assume that key being null is a programmer error, not a runtime error.